src/components/widgets-settings/params-picker/params-picker.spec.js

```diff
@@ -0,0 +1,70 @@
+describe('<> widget-setting-params-picker', function () {
+  'use strict';
+
+  var subject, subjectScope, $templateCache, $filter, ImpacWidgetsSvc, $httpBackend, $timeout;
+
+  // Compile the directive and return its scope
+  var compile = function() {
+    inject(function ($rootScope, $compile, $q) {
+      var scope = $rootScope.$new();
+
+      // Attributes passed to the directive
+      scope.widget = {settings: []};
+      scope.deferred = $q.defer();
+      scope.hasReach = true;
+      scope.param = "status_selection";
+
+      // Stubs
+      spyOn(scope.deferred, 'resolve').and.callThrough();
+
+      $compile(subject)(scope);
+    });
+    return subject.isolateScope();
+  };
+
+  beforeEach(function() {
+    module('maestrano.impac');
+
+    subject = angular.element('<div setting-params-picker options="statusOptions" has-reach="hasReach" param="status_selection" parent-widget="widget" class="part" deferred="::deferred" />');
+
+    // Inject services called by the directive...
+    inject(function (_ImpacWidgetsSvc_) {
+      ImpacWidgetsSvc = _ImpacWidgetsSvc_;
+    });
+
+    // ...and stub their methods
+    spyOn(ImpacWidgetsSvc, 'updateWidgetSettings').and.stub();
+  });
+
+  describe('.link', function(){
```

Comment:
#setting.toMetadata() is not tested: we should test that the returned hash corresponds to the attributes that have been passed to the directive, correctly formatted.